### PR TITLE
Allow for the injector to be built in docker

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This PR allows building the injector on more systems and migrates its release to GHCR to avoid issues with Docker Hub.

## Related Issue

Fixes #N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/injector/Makefile
Comment (Noxsios) [inline src/injector/Makefile:1]: might be good to do an `du --si` after building to make sure the resulting binary is < 1MiB

File: src/injector/Makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	rustup target add x86_64-unknown-linux-musl
+
+	if [ "$(shell uname -m)" = "arm64" ] || [ "$(shell uname -m)" = "aarch64" ]; then \
+		test -s x86_64-linux-musl-cross || curl https://musl.cc/x86_64-linux-musl-cross.tgz | tar -xz; \
+		export PATH="$$PWD/x86_64-linux-musl-cross/bin:$$PATH"; \
+		cargo build --target x86_64-unknown-linux-musl --release; \
+		x86_64-linux-musl-strip target/x86_64-unknown-linux-musl/release/zarf-injector; \
+	elif [ "$(shell uname -m)" = "x86_64" ]; then \
+		cargo build --target x86_64-unknown-linux-musl --release; \
+        strip target/x86_64-unknown-linux-musl/release/zarf-injector; \
```
Comment (Noxsios) [inline src/injector/Makefile:25]: strip should not be needed as `cargo.toml` is already configured to do so. https://github.com/johnthagen/min-sized-rust

Comment (Racer159) [reply src/injector/Makefile]: done

File: src/injector/Makefile
Comment (Racer159) [reply src/injector/Makefile:1]: done

**Comment by cmwylie19** [comment]
LGTM

**Review by YrrepNoj — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -151,6 +151,9 @@ build-examples: ## Build all of the example packages
 
 	@test -s ./build/zarf-package-yolo-$(ARCH).tar.zst || $(ZARF_BIN) package create examples/yolo -o build -a $(ARCH) --confirm
 
+build-injector-linux: ## Build the Zarf injector for AMD64 and ARM64
+	docker run --rm --user "$(id -u)":"$(id -g)" -v $$PWD/src/injector:/usr/src/zarf-injector -w /usr/src/zarf-injector rust:1.71.0-bookworm make build-injector-linux
+
```
Comment (cmwylie19) [inline Makefile:156]:
```suggestion
build-injector-linux: ## Build the Zarf injector for AMD64 and ARM64
	docker run --rm --user "$(id -u)":"$(id -g)" -v $PWD/src/injector:/usr/src/zarf-injector -w /usr/src/zarf-injector rust:1.71.0-bookworm make build-injector-linux

```

Comment (cmwylie19) [reply Makefile:156]: seems to be an extra $ in there

**Review by cmwylie19 — APPROVED** [review]
tested the make and it looks good from ARM64. I think this one lgtm

**Pull request merged**